app/services/payment_providers/moneyhash/customers/create_service.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module PaymentProviders
+  module Moneyhash
+    module Customers
+      class CreateService < BaseService
+        def initialize(customer:, payment_provider_id:, params:, async: true)
+          @customer = customer
+          @payment_provider_id = payment_provider_id
+          @params = params || {}
+          @async = async
+
+          super
+        end
+
+        def call
+          provider_customer = PaymentProviderCustomers::MoneyhashCustomer.find_by(customer_id: customer.id)
+          provider_customer ||= PaymentProviderCustomers::MoneyhashCustomer.new(customer_id: customer.id, payment_provider_id:)
+
+          if params.key?(:provider_customer_id)
+            provider_customer.provider_customer_id = params[:provider_customer_id].presence
+          end
+
+          if params.key?(:sync_with_provider)
+            provider_customer.sync_with_provider = params[:sync_with_provider].presence
+          end
+
+          provider_customer.save!
+
+          result.provider_customer = provider_customer
+
+          if should_create_provider_customer?
+            create_customer_on_provider_service(async)
+          elsif should_generate_checkout_url?
+            generate_checkout_url(async)
+          end
+
+          result
+        rescue ActiveRecord::RecordInvalid => e
+          result.record_validation_failure!(record: e.record)
+        end
+
+        private
+
+        attr_accessor :customer, :payment_provider_id, :params, :async
+
+        delegate :organization, to: :customer
+
+        def create_customer_on_provider_service(async)
+          return PaymentProviderCustomers::MoneyhashCreateJob.perform_later(result.provider_customer) if async
+
+          PaymentProviderCustomers::MoneyhashCreateJob.perform_now(result.provider_customer)
+        end
+
+        def generate_checkout_url(async)
+          return PaymentProviderCustomers::MoneyhashCheckoutUrlJob.perform_later(result.provider_customer) if async
```

Comment:
I don't understand what happens here if it's async.

When it's synchronous, the checkout URL is generated and returned in the API response so you can redirect your customer. But if it's async, what do you do with URL? It doesn't seem saved anywhere.